xdsl/ir.py

```diff
@@ -204,9 +204,13 @@ def owner(self) -> Operation:
         return self.op
 
     def __repr__(self) -> str:
-        return f"OpResult(typ={repr(self.typ)}, num_uses={repr(len(self.uses))}, " + \
-               f"op_name={repr(self.op.name)}, " + \
-               f"result_index={repr(self.result_index)}, name={repr(self.name)})"
+        return "{}<%{}: {}>(num_uses={}, op_name={})".format(
+            self.__class__.__name__,
+            self.name or self.result_index,
```

Comment:
I like the format, but I'm not sure about the self.result_index.
I'm mostly worried about cases where %{result_index} might collide with another %{self.name}, which could be confusing.
Could we use another prefix for this case ? Wondering what others think.

Reply:
Yeah I agree on this.

Reply:
result index can never collide with a name though, right? because you can't have names end in numbers? Or am I misunderstanding something?